main.go

```diff
@@ -26,6 +26,13 @@ func main() {
 	// Add Linode-specific flags
 	command.Flags().BoolVar(&linode.Options.LinodeGoDebug, "linodego-debug", false, "enables debug output for the LinodeAPI wrapper")
 
+	// Make the Linode-specific CCM bits aware of the kubeconfig flag
+	linode.Options.KubeconfigFlag = command.Flags().Lookup("kubeconfig")
```

Comment:
Unmanaged clusters will not typically provide this flag.

Reply:
We can allow it to be set to nil, and the check which falls through to InClusterConfig can check for `nil` instead of `""`

Reply:
In this case `nil` means that the flag exists at all, not that it was set. This flag value should be non-`nil` regardless of whether the CCM is started with the `--kubeconfig` flag (though I will grant that looking up the flag _structure_ vs. looking up the flag value is confusing).

Lookup docs: https://godoc.org/github.com/spf13/pflag#FlagSet.Lookup
Lookup source: https://github.com/spf13/pflag/blob/master/flag.go#L348-L350

If you dig through the source, you'll see that the various functions to define flags call `AddFlag`, which adds them to the `formal` map of flag objects before parsing.